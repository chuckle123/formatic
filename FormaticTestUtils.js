import React from 'react';
import { mount } from 'enzyme';

import Formatic from './lib/formatic';

export const renderFieldsToHtml = fields => {
  const config = Formatic.createConfig();
  const value = config.createRootValue({
    fields,
  });

  const component = mount(
    <Formatic
      config={config}
      value={value}
      fields={fields}
      onChange={() => {}}
    />
  );
  const html = component.html();
  component.unmount();
  return html;
};
