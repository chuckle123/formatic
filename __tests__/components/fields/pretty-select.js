/*global describe, it, expect*/

import { renderFieldsToHtml } from '../../../FormaticTestUtils';
import prettySelectExample from '../../../demo/examples/pretty-select';

describe('pretty-select field', () => {
  it('should render correctly', () => {
    expect(renderFieldsToHtml(prettySelectExample.fields)).toMatchSnapshot();
  });
});
