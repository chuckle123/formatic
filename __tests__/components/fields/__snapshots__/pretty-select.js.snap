// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pretty-select field should render correctly 1`] = `

<div class="formatic">
  <div class="optional">
    <div class>
      <span class="required-label not-required-text">
      </span>
    </div>
    <span>
      <span>
      </span>
      <fieldset class>
        <div class="optional">
          <div class>
            <label>
              Size
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                          <span>
                            Small
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="optional readonly">
          <div class>
            <label>
              Readonly Size
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                          <span>
                            Small
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="validation-error-required required">
          <div class>
            <label>
              Colors (R)
            </label>
            <span class="required-label required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box placeholder"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                          Pick a color...
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="optional">
          <div class>
            <label>
              Colors 2
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box placeholder"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="optional">
          <div class>
            <label>
              Colors 3
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <div class>
              Tres colores
            </div>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                          <span>
                            some default value
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="optional">
          <div class>
            <label>
              Alphabet
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box placeholder"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="optional">
          <div class>
            <label>
              Loading
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box placeholder"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="optional">
          <div class>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box placeholder"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
        <div class="optional">
          <div class>
            <label>
              Select with No Search
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div role="listbox"
                 tabindex="0"
                 class
            >
              <div>
                <div role="presentation"
                     class="pretty-text-wrapper"
                >
                  <div class="pretty-text-click-wrapper"
                       tabindex="0"
                       role="textbox"
                  >
                    <div role="presentation"
                         class="pretty-text-box"
                    >
                      <div class="internal-text-wrapper">
                        <span>
                          <span>
                            Small
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="JavaScript:"
                     class
                  >
                    Insert...
                  </a>
                </div>
                <span class="select-arrow">
                </span>
              </div>
              <span>
              </span>
            </div>
          </span>
        </div>
      </fieldset>
    </span>
  </div>
</div>

`;
